src/d3/line-chart.d3.ts

```diff
@@ -1,27 +1,102 @@
 import { Datum, XYChartD3 } from './xy-chart.d3';
 import * as d3 from 'd3';
+import { Observable } from 'rxjs';
+import { AUDIFICATION, t } from '../assets/i18n';
 
 export class LineChartD3 extends XYChartD3 {
-  protected appendChart(
+  activeLabelId = this.createId('active-label');
+  activeGroupId = this.createId('active-group');
+
+  protected renderData(
     svg: d3.Selection<SVGSVGElement, unknown, null, undefined>,
-    data: Datum[],
+    dataObservable: Observable<Datum[]>,
     scaleX: d3.ScaleTime<number, number>,
     scaleY: d3.ScaleLinear<number, number>,
-  ): void {
+  ) {
     const line = d3
       .line<Datum>()
       .defined(d => !isNaN(d.value))
       .x(d => scaleX(d.date))
       .y(d => scaleY(d.value));
 
-    svg
+    const path = svg
       .append('path')
-      .datum(data)
       .attr('fill', 'none')
       .attr('stroke', 'steelblue')
       .attr('stroke-width', 1.5)
       .attr('stroke-linejoin', 'round')
-      .attr('stroke-linecap', 'round')
-      .attr('d', line);
+      .attr('stroke-linecap', 'round');
+
+
+    const dataSubscription = dataObservable.subscribe(data => {
+      path
+        .datum(data)
+        .transition(this.transition)
+        .attr('d', line);
+    });
+
+    return () => {
```

Comment:
Would it make sense to split up the rendering of data and the removal of the rendered data into two methods? Then the base chart could handle all pipes / subscriptions to the dataObservable and the destroy methods could be called on finalize() or something like that? I think it would reduce complexity if we could limit the subscriptions to the dataObservable.

Reply:
It does look much less complex after splitting them up, thanks! Indeed, there is nothing much to tear down since utilizing the destroy$ pattern.